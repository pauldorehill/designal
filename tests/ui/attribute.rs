use designal::Designal;
use serde::{Deserialize, Serialize};

#[derive(Designal, Deserialize, Serialize)]
#[designal(trim_end = "Bean")]
#[designal(attribute =
    #[derive(Deserialize, Serialize)],
    #[serde(rename = "RenamedBean")]
)]
#[serde(rename = "RenamedBean")]
struct HumanBean {
    #[designal(attribute = #[serde(rename = "designal_new_name")])]
    #[serde(rename = "new_name")]
    taste: String,
}

fn main() {}