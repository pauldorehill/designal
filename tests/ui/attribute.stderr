error: duplicate serde attribute `rename`
  --> $DIR/attribute.rs:10:9
   |
10 | #[serde(rename = "RenamedBean")]
   |         ^^^^^^

error: duplicate serde attribute `rename`
  --> $DIR/attribute.rs:13:13
   |
13 |     #[serde(rename = "new_name")]
   |             ^^^^^^
